flow/hooks/log_operations.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2018 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Builtin execution hook for basic logging."""
+import logging
+
+
+class LogOperations:
+    """:class:`~.LogOperations` logs information about the execution of operations to a log file.
+
+    This hooks provides information, optionally, on the start, successful completion, and/or
+    erroring of one or more operations in a `flow.FlowProject` instance. The logs are stored in a
+    file given by the parameter ``fn_logfile``. This file will be appended to if it already exists.
+
+    The default formating for the log provides the time, job id, log level, and log message.
+
+    Examples
+    --------
+    The code block below provides an example of how install  :class:`~.LogOperations` to a
+    instance of :class:`~.FlowProject`
+
+    .. code-block:: python
+        from flow import FLowProject
+        from flow.hooks import LogOperations  # Import build
+
+
+    Parameters
+    ----------
+    fn_logfile: log filename
+        The name of the log file in the job workspace. Default is "execution-record.log".
+    """
+
+    def __init__(self, fn_logfile="execution-record.log"):
+        self._fn_logfile = fn_logfile
+        # getLogger keep its own cache. This just serves to reduce the time spent setting up loggers
+        # by only doing it once.
+        self._loggers = dict()
```

Comment:
I think its fine this way but sometimes linters / type checkers seem to complain about it? 

```suggestion
        self._loggers = {}
```